pkg/inventory/db.go

```diff
@@ -176,7 +177,8 @@ func (db *DB) Run(ctx context.Context) error {
 		}
 
 		klog.V(4).Infof("Found %d devices", len(devices))
-		if len(devices) > 0 {
+		if len(devices) > 0 || db.hasDevices {
+			db.hasDevices = len(devices) > 0
```

Comment:
This might be racy. Please check.

Reply:
I do not see a concurrent access to the boolean